File: src/types/ExportQuality.ts
export type ExportQuality = 'standard' | 'high';

export interface QualityProfile {
  readonly maxStillPixelWidth: number;
  readonly jpegCompression: number;
  readonly videoBitrate: number;
}

export const qualityProfile = (q: ExportQuality): QualityProfile => {
  if (q === 'high') {
    return { maxStillPixelWidth: 2160, jpegCompression: 0.95, videoBitrate: 8_000_000 };
  }
  return { maxStillPixelWidth: 1280, jpegCompression: 0.85, videoBitrate: 3_500_000 };
};

File: _legacy-swift/Models/ExportQuality.swift
import AVFoundation
import Foundation

/// Target export quality for the generated Live Photo pair.
public enum ExportQuality: String, Equatable, Sendable {
    case standard
    case high

    /// AVAssetExportSession preset for the MOV component.
    public var assetExportPreset: String {
        switch self {
        case .standard:
            return AVAssetExportPreset1280x720
        case .high:
            return AVAssetExportPresetHighestQuality
        }
    }

    /// Target still-image pixel width for the JPEG component.
    public var maxStillPixelWidth: CGFloat {
        switch self {
        case .standard:
            return 1280
        case .high:
            return 2160
        }
    }

    /// JPEG compression quality (0.0 - 1.0).
    public var jpegCompression: CGFloat {
        switch self {
        case .standard:
            return 0.85
        case .high:
            return 0.95
        }
    }
}
